Classes/Ajax/Dispatcher.php

```diff
@@ -121,10 +111,12 @@ protected function initTYPO3() {
 			'sys_language_overlay' => 'hideNonTranslated',
 			'sys_language_softMergeIfNotBlank' => '',
 			'sys_language_softExclude' => '',
-			'language' => $lang,
+			'language' => $GLOBALS['TSFE']->tmpl->setup['config.']['language'],
+            'locale_all' => $GLOBALS['TSFE']->tmpl->setup['config.']['locale_all'],
```

Comment:
When `config.language` or `config.locale_all` is not set in TypoScript, this will result in an "Undefined index" PHP notice. Also, the indentation looks odd, here.